File: Python/MaximumProductSubArray.py
def maxProductSubArray(nums):
    result = max(nums)
    curMax = 1
    curMin = 1

    for n in nums:
        tempMax = n*curMax
        curMax = max(n,n*curMax,n*curMin)
        curMin = min(n,n*curMin,tempMax)
        result = max(result,curMax)

    return result

File: Javascript/MaximumProductSubArray.js
var maxProductSubArray = function(nums){
    let result = Math.max(...nums)
    let curMax = 1
    let curMin = 1
    
    for (const n of nums){
        let tempMax = n*curMax
        curMax = Math.max(n,n*curMax,n*curMin)
        curMin = Math.min(n,n*curMin,tempMax)
        result = Math.max(result,curMax)
    }

    return result
}